.github/workflows/test-appetize.yml

```diff
@@ -5,15 +5,45 @@ on:
   pull_request:
     types: [labeled, synchronize]
     branches:
+      - deployment/*
       - master
 jobs:
+  set_environment:
+    if: contains(github.event.pull_request.labels.*.name, 'Test - appetize')
+    outputs:
+      my_env: ${{ steps.setenv.outputs.MY_ENV }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set Environment Name
+      run: |
+        if [ "$GITHUB_BASE_REF" = "master" ];                    \
+        then echo "ENVIRONMENT_NAME=plh_global" >> $GITHUB_ENV;              \
+        else echo "ENVIRONMENT_NAME=${GITHUB_BASE_REF##*/}" >> $GITHUB_ENV;  \
+        fi
+    - id: setenv
+      run: echo "MY_ENV=${ENVIRONMENT_NAME}" >> $GITHUB_OUTPUT
   test_appetize:
+    needs: set_environment
     if: contains(github.event.pull_request.labels.*.name, 'Test - appetize')
     runs-on: ubuntu-latest
+    environment: 
+      name: ${{ needs.set_environment.outputs.MY_ENV }}
     steps:
       - uses: actions/checkout@v3
         with:
           lfs: true
+      # Extract deployment name as the slug of target PR branch via string operators, e.g. /deployment/plh_tz -> plh_tz
+      # https://www.linuxjournal.com/article/8919, https://github.com/greenbone/gsa/actions/runs/1277581365/workflow
+      - name: Set Environment Deployment
+        run: |
+          if [ "$GITHUB_BASE_REF" = "master" ];                    \
```

Comment:
nit(blocking) 
Do we need to be repeating this? I thought it was already calculated during the `set_environment` job?